trento/migration/sso-integration.md

```diff
@@ -0,0 +1,441 @@
+   -e OAUTH2_TOKEN_URL=<OAUTH2_TOKEN_URL> \
+   -e OAUTH2_USER_URL=<OAUTH2_USER_URL> \
+
+   # Optional:
+   -e OAUTH2_SCOPES=<OAUTH2_SCOPES> \
+   -e OAUTH2_CALLBACK_URL=<OAUTH2_CALLBACK_URL> \
+
+   ...[other settings]...
+   ```
+
+### Available variables for OAuth 2.0
+
+OAUTH2_CLIENT_ID
+
+: OAUTH2 client id
+
+OAUTH2_CLIENT_SECRET
+
+: OAUTH2 client secret
+
+OAUTH2_BASE_URL
+
+: OAUTH2 base url
+
+OAUTH2_AUTHORIZE_URL
+
+: OAUTH2 authorization url
+
+OAUTH2_TOKEN_URL
+
+: OAUTH2 token url
+
+OAUTH2_USER_URL
+
+: OAUTH2 token url
+
+OAUTH2_SCOPES
+
+: OAUTH2 scopes, used to define the user values sent to the SP. It must be adjusted depending on IDP provider requirements (default value: `openid profile email`)
+
+OAUTH2_CALLBACK_URL
+
+: OAUTH2 callback url where the IDP is redirecting once the authentication is completed (default value: `https://#{TRENTO_WEB_ORIGIN}/auth/oauth2_callback`)
+
+
+## Using SAML
+
+Trento integrates with an IDP that uses the SAML protocol to authenticate users accessing the Trento web console. Trento will behave as a Service Provider (SP) in this case.
+
+Commonly, SAML protocol messages are signed with SSL. This is optional using Trento, and the signing is not required (even though it is recommended).
+If the IDP signs the messages, and expect signed messages back, certificates used by the SP (Trento in this case) must be provided to the IDP, the public certificate file in this case.
+
+To use an existing SAML IDP, follow the next instrunctions to met the specific requirements. You need:
+
+* Configure SAML IDP and user profiles
+* Start Trento to generate the certificates and download them
+* Provide the generated certificate to the IDP
+* Restart Trento
+
+### Configuring SAML IDP setup
+
+Configure the existing IDP with the next minimum options to be able to connect with Trento's Service Provider (SP).
+
+#### SAML user profile
+
+Users provided by the SAML installation must have some few mandatory attributes to login in Trento. All of them are mandatory, even though their name is configurable.
+The user profile must include attributes for: username, email, first name and last name.
+This attributes must be mapped for all users wanting to connect to Trento.
+
+By default, Trento expects the `username`, `email`, `firstName` and `lastName` attribute names. All these 4 attribute names are configurable using the next environment variables, following the same order: `SAML_USERNAME_ATTR_NAME`, `SAML_EMAIL_ATTR_NAME`, `SAML_FIRSTNAME_ATTR_NAME` and `SAML_LASTNAME_ATTR_NAME`.
+So for example, if the IDP user profile username is defined as `attr:username`, `SAML_USERNAME_ATTR_NAME=attr:username` should be used.
+
+#### SAML redirect URI
+
+Once the login is done succesfully, the IDP redirects the session back to Trento. This redirection is done to `https://trento.example.com/sso/sp/consume/saml`, so this URI must be set as valid in the IDP.
+
+### Get certificates from Trento
```

Comment:
```suggestion
### Downloading certificates from Trento
```

Reply:
I'm going with `Getting` as you get the content as a string, not a file itself